OpenRA.Mods.Common/Traits/Modifiers/FrozenUnderFog.cs

```diff
@@ -62,10 +62,26 @@ public FrozenUnderFog(ActorInitializer init, FrozenUnderFogInfo info)
 			startsRevealed = info.StartsRevealed && !init.Contains<ParentActorInit>();
 			var footprintCells = FootprintUtils.Tiles(init.Self).ToList();
 			footprint = footprintCells.SelectMany(c => map.ProjectedCellsCovering(c.ToMPos(map))).ToArray();
+
 			tooltip = Exts.Lazy(() => init.Self.TraitsImplementing<ITooltip>().FirstOrDefault());
 			health = Exts.Lazy(() => init.Self.TraitOrDefault<Health>());
 		}
 
+		public void AddedToWorld(Actor self)
+		{
+			if (initialized)
+				return;
```

Comment:
If this is removed and circumstances change, then that bug would be exposed. One easy case for exposing such is to give an actor both `FrozenUnderFog` and `Passenger` traits such as a mod or a map that uses `FrozenUnderFog` for showing a last seen location of a transportable actor.